includes/class-wc-stripe-blocks-support.php

```diff
@@ -60,6 +60,63 @@ public function is_active() {
 	 * @return array
 	 */
 	public function get_payment_method_script_handles() {
+		if ( WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
+			$this->register_upe_payment_method_script_handles();
+		} else {
+			$this->register_classic_payment_method_script_handles();
+		}
+
+		return [ 'wc-stripe-blocks-integration' ];
+	}
+
+	/**
+	 * Registers the UPE JS scripts.
+	 */
+	private function register_upe_payment_method_script_handles() {
+		$asset_path   = WC_STRIPE_PLUGIN_PATH . '/build/upe_blocks.asset.php';
+		$version      = WC_STRIPE_VERSION;
+		$dependencies = [];
+		if ( file_exists( $asset_path ) ) {
+			$asset        = require $asset_path;
+			$version      = is_array( $asset ) && isset( $asset['version'] )
+				? $asset['version']
+				: $version;
+			$dependencies = is_array( $asset ) && isset( $asset['dependencies'] )
+				? $asset['dependencies']
+				: $dependencies;
+		}
+
+		// Manually register stripe since it's missing for UPE checkout for some reason.
+		wp_register_script(
+			'stripe',
+			'https://js.stripe.com/v3/',
+			[],
+			'3.0',
+			true
+		);
+
+		$dependencies[] = 'stripe';
+
+		wp_register_script(
+			'wc-stripe-blocks-integration',
+			WC_STRIPE_PLUGIN_URL . '/build/upe_blocks.js',
+			$dependencies,
+			$version,
+			true
+		);
+		wp_set_script_translations(
+			'wc-stripe-blocks-integration',
+			'woocommerce-gateway-stripe'
+		);
+
+		// TODO: remove once script conflict has been resolved.
+		wp_dequeue_script( 'wc-stripe-upe-classic' );
```

Comment:
I'm hoping we can remove this once #1855 has been merged since the classic UPE JS (`upe_classic.js`) has some sort of conflict with the Blocks integration (`upe_blocks.js`).

**Note:** this (very simple) workaround breaks the shortcode checkout flow, so we should not merge this PR until this has been resolved.

Reply:
This has been removed since #1855 has been merged.

The issue itself also turned out to be a bit more nuanced, as can be seen with https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1825/commits/a9774f9f6745f8effaa9d3fb0df6d133248999fa